index.html

```diff
@@ -1147,59 +1205,81 @@
       </section>
       <section>
         <h2>
-          The "<dfn>pushsubscriptionchange</dfn>" event
+          The <dfn>pushsubscriptionchange</dfn> event
         </h2>
         <p>
-          The <a>pushsubscriptionchange</a> event indicates that a <a>push subscription</a>
-          has been invalidated, or will soon be invalidated. For example, the <a>push service</a>
-          MAY set an expiration time. A <a>Service Worker</a> SHOULD attempt to resubscribe while
-          handling this event, in order to continue receiving <a>push messages</a>.
+          The <a>pushsubscriptionchange</a> event indicates a change in a <a>push subscription</a>
+          that was triggered outside of the application's control, for example because it has been
+          refreshed, revoked or lost.
         </p>
         <p>
-          When new <a>push subscription</a> information becomes available, the <a>user agent</a>
-          MUST run the following steps:
+          To <dfn>fire the pushsubscriptionchange event</dfn> given a
+          <a>service worker registration</a> of <var>registration</var>, <var>newSubscription</var>
+          and <var>oldSubscription</var>, the <a>user agent</a> must run the following steps:
         </p>
         <ol>
-          <li>Let <var>registration</var> be the <a>service worker registration</a> corresponding
-          to the push message.
+          <li>Create a <a>trusted event</a>, <var>event</var>, that uses the
+          <a>PushSubscriptionChangeEvent</a> interface, with the event type
+          <code>pushsubscriptionchange</code>, which does not bubble, is not cancelable, and has no
+          default action.
           </li>
-          <li>If <var>registration</var> is not found, abort these steps.
+          <li>Set the <code>newSubscription</code> attribute of <var>event</var> to
+          <var>newSubscription</var>.
           </li>
-          <li>Invoke the <a>Handle Functional Event</a> algorithm with a <a>service worker
-          registration</a> of <var>registration</var> and <var>callbackSteps</var> set to the
-          following steps:
+          <li>Set the <code>oldSubscription</code> attribute of <var>event</var> to
+          <var>oldSubscription</var>.
+          </li>
+          <li>Invoke the <a>Handle Functional Event</a> algorithm with <var>event</var>,
+          <var>registration</var> and <var>callbackSteps</var> set to the following steps:
```

Comment:
The list construction implies that all three inputs are set to the following steps.  I don't know how to improve that with some more restructuring though.